pkgs/native_assets_cli/example/build/local_asset/test/build_test.dart

```diff
@@ -12,10 +11,16 @@ void main() {
   testBuildHook(
     description: 'test my build hook',
     mainMethod: build.main,
+    extraConfigSetup: (config) {
+      config.setupCodeConfig(
```

Comment:
> Can we use the builders for all instead?

We do use builders for all - the entire config is built using `{Build,Link}ConfigBuilder.setup*` methods.

> I think it's a bit smelly that

It's just a layered architecture:

* "base protocol requirements" are initialized on the builder in `package:native_assets_builder`: right now `taretOS`, `linkingEnabled`  - but also things determined internally such as the `outputDirectory`, ...
* "extensible things" are initialized in the bundling tool (e.g. code asset configuration, code asset type, code asset encoding)

Some of these probably shouldn't be in the "base protocol requirements" (like `targetOS`) but that's a separate discussion.

> Or did you want to avoid modifying the json serialization of "base config" parts?

It's just symmetric with the remaining layering. The `package:native_assets_builder` is responsible for the base protocol things and the bundling tool for any extensible things.

> Or instead of trying to go builders all the way, we could go the other way and introduce a code_assets/code_asset_test.dart with testCodeBuildHook that takes a flat list of parameters.

Yes. Each asset type can have their own test wrapper. (as I mentioned the PR isn't quite done yet)

Reply:
> Or instead of ... we could go the other way and introduce a code_assets/code_asset_test.dart with testCodeBuildHook that takes a flat list of parameters

Done